sdk/examples/xo_go/src/sawtooth_xo/handler/handler.go

```diff
@@ -0,0 +1,321 @@
+/**
+ * Copyright 2017 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------------
+ */
+
+package handler
+
+import (
+	"bytes"
+	"crypto/sha512"
+	"encoding/hex"
+	"fmt"
+	"github.com/golang/protobuf/proto"
+	"sawtooth_sdk/logging"
+	"sawtooth_sdk/processor"
+	"sawtooth_sdk/protobuf/processor_pb2"
+	"sawtooth_sdk/protobuf/transaction_pb2"
+	"strconv"
+	"strings"
+)
+
+var logger *logging.Logger = logging.Get()
+var namespace = hexdigest("xo")[:6]
+
+type XoPayload struct {
+	Name   string
+	Action string
+	Space  int
+}
+
+type XoHandler struct {
+}
+
+type Game struct {
+	Board   string
+	State   string
+	Player1 string
+	Player2 string
+	Name    string
+}
+
+func (self *XoHandler) FamilyName() string {
+	return "xo"
+}
+func (self *XoHandler) FamilyVersion() string {
+	return "1.0"
+}
+func (self *XoHandler) Encoding() string {
+	return "csv-utf8"
+}
+
+func (self *XoHandler) Namespaces() []string {
+	return []string{namespace}
+}
+
+func (self *XoHandler) Apply(request *processor_pb2.TpProcessRequest, state *processor.State) error {
+	// The xo player is defined as the signer of the transaction, so we unpack
+	// the transaction header to obtain the signer's public key, which will be
+	// used as the player's identity.
+	header, err := unpackHeader(request.Header)
+	if err != nil {
+		return err
+	}
+	player := header.SignerPubkey
```

Comment:
My convention has been to use the helper methods here, so instead I would do `header.GetSignerPubkey()`.

"There are getters that return a field's value if set, and return the field's default value if unset. The getters work even if the receiver is a nil message." -https://github.com/golang/protobuf